src/engagement.ts

```diff
@@ -0,0 +1,258 @@
+      items,
+      totalItems: items.length
+    }
+  } catch (error) {
+    core.warning(`Failed to get issues: ${error}`)
+    return {
+      items: [],
+      totalItems: 0
+    }
+  }
+}
+
+/**
+ * Get detailed information about an issue including comments
+ */
+async function getIssueDetails(
+  octokit: ReturnType<typeof github.getOctokit>,
+  owner: string,
+  repo: string,
+  issueNumber: number
+): Promise<IssueDetails> {
+  const { data: issue } = await octokit.rest.issues.get({
+    owner,
+    repo,
+    issue_number: issueNumber
+  })
+
+  // Get issue comments
+  const { data: comments } = await octokit.rest.issues.listComments({
+    owner,
+    repo,
+    issue_number: issueNumber
+  })
+
+  const commentsData: CommentData[] = comments.map((comment) => ({
+    id: comment.id,
+    user: comment.user!,
+    created_at: comment.created_at,
+    reactions: comment.reactions!
+  }))
+
+  return {
+    id: issue.id,
+    number: issue.number,
+    title: issue.title,
+    body: issue.body || '',
+    state: issue.state,
+    created_at: issue.created_at,
+    updated_at: issue.updated_at,
+    closed_at: issue.closed_at,
+    comments: issue.comments,
+    reactions: issue.reactions!,
+    comments_data: commentsData,
+    user: issue.user!,
+    assignees: issue.assignees!
+  }
+}
+
+/**
+ * Calculate engagement score for an issue based on the engagement algorithm
+ */
+function calculateScore(issue: IssueDetails): number {
+  // Components:
+  // - Number of Comments       => Indicates discussion and interest
+  // - Number of Reactions      => Shows emotional engagement
+  // - Number of Contributors   => Reflects the diversity of input
+  // - Time Since Last Activity => More recent activity indicates higher engagement
+  // - Issue Age                => Older issues might need more attention
+  // - Number of Linked PRs     => Shows active work on the issue (not implemented)
+
+  const totalComments = issue.comments
+  const totalReactions =
+    issue.reactions.total_count +
+    (issue.comments_data?.reduce((sum, comment) => sum + comment.reactions.total_count, 0) || 0)
+  const contributors = getUniqueContributors(issue)
+  const lastActivity = Math.max(1, getTimeSinceLastActivity(issue))
+  const issueAge = Math.max(1, getIssueAge(issue))
+  const linkedPullRequests = 0 // Not implemented yet
+
+  // Weights:
+  const CommentsWeight = 3
+  const ReactionsWeight = 1
+  const ContributorsWeight = 2
+  const LastActivityWeight = 1
+  const IssueAgeWeight = 1
+  const LinkedPullRequestsWeight = 2
+
+  const score =
+    CommentsWeight * totalComments +
+    ReactionsWeight * totalReactions +
+    ContributorsWeight * contributors +
+    LastActivityWeight * (1 / lastActivity) +
+    IssueAgeWeight * (1 / issueAge) +
+    LinkedPullRequestsWeight * linkedPullRequests
+
+  return Math.round(score)
+}
+
+/**
+ * Calculate previous score (7 days ago) - simplified implementation
+ */
+function calculatePreviousScore(): number {
```

Comment:
Please fully implement this. Make sure to add tests

Look at all these tests and make sure to add all of these https://github.com/mattleibow/dev-tools/tree/main/LabeledByAI.Services.Tests/Services/Engagement

Look at all the files in there and make sure you have all those tests in this repo.